### PR TITLE
Add execution-level CUDA Graph support for 2.0

### PR DESCRIPTION
**Pull request by AllenYolk**
## Summary

- add opt-in manual CUDA Graph execution to Common and single-rank Vision training/inference
- integrate MCore native CUDA Graph lifecycle and keep SGLang graphs disabled until temporal metadata is supported
- add eager/compile/manual-graph benchmark reporting and CUDA Graph design documentation
- set the package version to 2.0.0
- remove a stale FlexSN probe test that asserted the pre-refactor eager kernel-construction lifecycle

## Validation

- `pytest -q test/activation_based`: 1392 passed, 386 skipped
- g3 CUDA regression (Torch 2.7.1, Triton 3.3.1, RTX 4090): 10 passed
- g3 full single-GPU benchmark tests: 20 passed
- Ruff checks passed
- Sphinx HTML build passed
- changelog generation check passed

## Summary by CodeRabbit

* **New Features**
  * Added optional CUDA Graph execution for Common and Vision training, evaluation, prediction, and benchmarking.
  * Added `eager`, `compile`, and `cuda_graph` execution modes with configurable warmup and performance statistics.

* **Bug Fixes**
  * Improved Triton LIF precision and reduced unnecessary voltage-sequence allocation.
  * Fixed surrogate-gradient precision and Triton backend configuration.

* **Breaking Changes**
  * Replaced Vision’s `compile` setting and Common Trainer’s `--compile` option with execution-mode configuration.

* **Documentation**
  * Documented CUDA Graph limitations, fallback behavior, and configuration.

* **Release**
  * Promoted the project version to 2.0.0.

### CODE REVIEW

File: benchmark/vision_distributed.py [hunk 1]
```diff
@@ -143,6 +143,12 @@ def _parse_args() -> argparse.Namespace:
     parser.add_argument("--max-steps", type=int, default=10)
     parser.add_argument("--scheduler-steps", type=int)
     parser.add_argument("--timing-warmup-steps", type=int, default=0)
+    parser.add_argument(
+        "--execution-mode",
+        choices=("eager", "compile", "cuda_graph"),
+        default="eager",
+    )
+    parser.add_argument("--cuda-graph-warmup-steps", type=int, default=11)
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply benchmark/vision_distributed.py]: Fixed in 9846f824. The benchmark override now defaults to 3 warmup steps, so the default 10-step CUDA Graph run reaches capture and replay. Verified on g3 with 1 capture, 6 replays, and 0 fallbacks.

File: docs/design/cudagraph.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- MCore owns its `cuda_graph_impl`, modules, and inference scope.
+- SGLang CUDA Graph remains disabled until its temporal adapters provide the
+  required attention metadata and pass native Engine capture/replay tests.
+
+## Supported first release
+
+- Common and Vision: CUDA, fixed shape/dtype/layout, Torch or Triton neuron
+  backend, `store_v_seq=False`, and single-rank execution. Distributed CUDA
+  Graph execution remains owned by MCore because the project's PyTorch
+  2.7/NCCL environment cannot capture collectives.
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply docs/design/cudagraph.md]: Partially accepted in 9846f824: the design now records the actual collective-capture failure environment (PyTorch 2.7.1+cu118, CUDA 11.8, NCCL 2.21.5). I did not duplicate the unrelated PyTorch 2.8 benchmark environment here; that remains scoped where those benchmark results are documented.

File: pyproject.toml [hunk 3]
```diff
@@ -10,7 +10,7 @@ include = ["spikingjelly*"]
 
 [project]
 name = "spikingjelly"
-version = "2.0.0rc1"
+version = "2.0.0"
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply pyproject.toml:13]: Not changed. Version 2.0.0 is the explicitly selected development target, while Unreleased correctly contains changes that have not yet been published. The entries should move to a dated section when the release is cut, not merely when development starts.

File: spikingjelly/activation_based/distributed/llm/training.py [hunk 4]
```diff
@@ -312,6 +337,12 @@ def train(
         training_seconds = 0.0
         timed_steps = 0
         for step in range(start_step + 1, config.train_steps + 1):
+            if (
+                cuda_graph_helper is not None
+                and not cuda_graph_helper.capture_finished()
+                and step - start_step - 1 == transformer.cuda_graph_warmup_steps
+            ):
+                cuda_graph_helper.create_cudagraphs()
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/llm/training.py:368]: Fixed in 9846f824. The transformer_engine path now initializes the pinned MCore 0.18.2 global microbatch calculator before constructing/capturing the helper and destroys it only when this training call initialized it.
